src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -49,51 +47,87 @@ read_aie_metadata(const char* data, size_t size, pt::ptree& aie_project)
   pt::read_json(aie_stream,aie_project);
 }
 
-std::vector<tile_type>
-get_tiles(const pt::ptree& aie_meta, const std::string& graph_name)
+adf::driver_config
+get_driver_config(const pt::ptree& aie_meta)
 {
-  std::vector<tile_type> tiles;
+    adf::driver_config driver_config;
+    driver_config.hw_gen = aie_meta.get<uint8_t>("aie_metadata.driver_config.hw_gen");
+    driver_config.base_address = aie_meta.get<uint64_t>("aie_metadata.driver_config.base_address");
+    driver_config.column_shift = aie_meta.get<uint8_t>("aie_metadata.driver_config.column_shift");
+    driver_config.row_shift = aie_meta.get<uint8_t>("aie_metadata.driver_config.row_shift");
+    driver_config.num_columns = aie_meta.get<uint8_t>("aie_metadata.driver_config.num_columns");
+    driver_config.num_rows = aie_meta.get<uint8_t>("aie_metadata.driver_config.num_rows");
+    driver_config.shim_row = aie_meta.get<uint8_t>("aie_metadata.driver_config.shim_row");
+    driver_config.reserved_row_start = aie_meta.get<uint8_t>("aie_metadata.driver_config.reserved_row_start");
+    driver_config.reserved_num_rows = aie_meta.get<uint8_t>("aie_metadata.driver_config.reserved_num_rows");
+    driver_config.aie_tile_row_start = aie_meta.get<uint8_t>("aie_metadata.driver_config.aie_tile_row_start");
+    driver_config.aie_tile_num_rows = aie_meta.get<uint8_t>("aie_metadata.driver_config.aie_tile_num_rows");
+    return driver_config;
+}
+
+adf::graph_config
+get_graph(const pt::ptree& aie_meta, const std::string& graph_name)
+{
+    adf::graph_config graph_config;
 
   for (auto& graph : aie_meta.get_child("aie_metadata.graphs")) {
     if (graph.second.get<std::string>("name") != graph_name)
       continue;
 
+    graph_config.id = graph.second.get<int>("id");
+    graph_config.name = graph.second.get<std::string>("name");
+
     int count = 0;
     for (auto& node : graph.second.get_child("core_columns")) {
-      tiles.push_back(tile_type());
-      auto& t = tiles.at(count++);
-      t.col = std::stoul(node.second.data());
+      graph_config.coreColumns.push_back(std::stoul(node.second.data()));
+      count++;
     }
 
     int num_tiles = count;
+
     count = 0;
     for (auto& node : graph.second.get_child("core_rows"))
-      tiles.at(count++).row = std::stoul(node.second.data());
+    {
```

Comment:
Please use the format like 
for () {
}

Reply:
done